### ISSUE TITLE
UX Trivia

### ISSUE THREAD

**Issue by greggman** [samhocevar/portable-file-dialogs#40]
Just passing on info, not a criticism.

It used to be Apple's UX Guidelines suggested not to use "Yes", "No", "Cancel" but instead the verbs of the action

Sn for example 

    "Save the file before closing the application"  (Save), (Quit without Saving), (Cancel)

etc.... 

"Yes", "No", "Cancel" and be very confusing. What does "No" mean to "Save the file before closing the application?".

* "No I don't want to save but I do want to quit", 
* "No I don't want to and no I don't want to quit",

From their [current guidelines](https://developer.apple.com/design/human-interface-guidelines/ios/views/alerts/)

> To the extent possible, use verbs and verb phrases that relate directly to the alert title and message—for example, View All, Reply, or Ignore. Use OK for simple acceptance. Avoid using Yes and No.

**Comment by merwok** [comment]
On that topic, Gnome and Gtk guidelines have a different order for buttons: `[Cancel]   [Save]`
Not sure which reference is the current one, but here’s one example: https://developer.gnome.org/hig-book/3.0/design-window.html.en

**Comment by samhocevar** [comment]
I agree that platform-specific UX recommendations should be followed whenever possible. One goal of PFD is to prevent programmers from rolling their own message boxes, file browsers, etc. that clash with the rest of the desktop experience, because, well, programmers are not necessarily UX designers. Using verbs for button labels makes a lot of sense and I’ll look into a reasonable API to support that.

Note that the GNOME backend automatically follows the guidelines for button positions, as can be seen in the [`pfd::message` screenshots](https://github.com/samhocevar/portable-file-dialogs/blob/master/doc/message.md#Screenshots)

A few links for future reference:
 - [rename buttons in KDialog](https://scottferguson.com.au/community/FOSS/kdialog.html)
 - [use a CBT hook to rename Windows MessageBox buttons](https://www.codeguru.com/cpp/w-p/win32/messagebox/article.php/c10873/MessageBox-with-Custom-Button-Captions.htm) (the Windows [`MessageBox` API](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-messageboxexa) does not provide an easy way to rename buttons)

**Comment by merwok** [comment]
The gnome screenshot doesn’t have the main positive action on the rightmost position.

**Comment by samhocevar** [comment]
You're right. This is a bit messed up.
